File: algorithms/scala/Fibonacci.scala
import scala.annotation.tailrec

/**
 * Find a number from the fibonacci sequence, in this cases the 1st through to the 7th.
 */
object Fibonacci {

  def main(args: Array[String]) = {
    println(fib(0))
    println(fib(1))
    println(fib(2))
    println(fib(3))
    println(fib(4))
    println(fib(5))
    println(fib(6))
  }

  /**
   * Get the fibonacci number for a given index.
   *
   * @param index the index in the fibonacci sequence to generate the fibonacci number for.
   * @return the generated number.
   */
  def fib(index: Long) = {
    @tailrec
    def fibRecursion(current: Long, prev: Long, prev2: Long): Long =
      if(current > 0) fibRecursion(current - 1, prev2, prev + prev2) else prev

    fibRecursion(index, 0, 1)
  }
}

File: algorithms/ruby/fibonacci.rb
module Sequences
  public

  # Get the fibonacci number for a given index.
  def fibonacci(num)
    return 1 if num <= 1
    fibonacci(num - 1) + fibonacci(num - 2)
  end
end

if __FILE__ == $0
  include Sequences

  # Find a number from the fibonacci sequence, in this cases the 1st through to the 8th.
  (0..7).each {|idx| puts fibonacci idx }
end